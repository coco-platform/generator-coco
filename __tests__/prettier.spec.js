/**
 * @description - coco:prettier generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const renderer = require('yeoman-test');
const assert = require('yeoman-assert');

describe('coco:prettier', () => {
  it('should generate prettier configuration', () => {
    const prettier = path.resolve(__dirname, '../prettier');

    return renderer.run(prettier).then(() => {
      assert.file(['.prettierrc']);

      const configuration = fs.readFileSync('.prettierrc', 'utf8');

      expect(configuration).toMatchSnapshot();
    });
  });
});
