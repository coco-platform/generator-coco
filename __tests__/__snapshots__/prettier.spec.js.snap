// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:prettier should generate prettier configuration 1`] = `
"{
  \\"singleQuote\\": true,
  \\"arrowParens\\": \\"always\\",
  \\"trailingComma\\": \\"es5\\"
}
"
`;
